### ISSUE TITLE
No mention on @using required to override default layout template

### ISSUE THREAD

**Issue by bmeenehan** [dotnet/AspNetCore.Docs#31170]
### Description

It is clear how to make a custom layout, using LayoutComponentBase and the @Body directive.
However, it is not mentioned that an @using directive is also required to use the @layout directive on a page.

### Page URL

https://learn.microsoft.com/en-us/aspnet/core/blazor/components/layouts?view=aspnetcore-8.0

### Content source URL

https://github.com/dotnet/AspNetCore.Docs/blob/main/aspnetcore/blazor/components/layouts.md

### Document ID

58e0022a-ac6c-9bf8-380a-40b7d068a887

### Article author

@guardrex

**Comment by guardrex** [comment]
Can you find the layout without using the namespace in your `_Imports` file by referecing it the same way at the project template with `Layout.` supplied ...

```razor
<RouteView RouteData="@routeData" DefaultLayout="@typeof(Layout.{CUSTOM LAYOUT})" />
```

**Comment by guardrex** [comment]
I'm going to open a product unit issue on this. I feel like they should've give the `Layout` folder an entry in the `_Imports` file. They might have had a reason not to do it, but they'll tell us that on the issue I'm about to open.

**Comment by bmeenehan** [comment]
Yes this did work, and I tried it in the testing process, but I want to
have more than one layout file, and for the MainLayout to remain the
default.
Is there a way to do that in the Routes file, and if so is that best
practice? Adding the @using directive is not a big deal, I simply had to
flail a little to land on that solution.

**Comment by guardrex** [comment]
I opened an issue to ask them about it :point_up:. If they say that they'll do it, then I'll try to address this with a general remark at the top of the article saying that the namespace of the layouts wherever they reside must be known and suggest that an `_Imports` file entry is a fine spot.

WRT your question ... if you just mean setting the layout on a per-component basis, you should be able to just name the layout at the top of a component with `@layout {LAYOUT TO USE}`.

... OR ...

You can add an `_Imports` file anywhere in the directory tree with the `@layout` directive in it, and it will apply that layout to everything at or below that level.

* Components
  * Pages
    * Page1.razor
    * Page2.razor
    * Admin
      * Page3.razor
      * Page4.razor
      * _Imports.razor
 
A layout applied by the `Imports` file in the `Admin` folder with `@layout` will only apply that layout to `Page3` and `Page4` components. The `Page1` and `Page2` components will get the default layout.

**Comment by guardrex** [comment]
I'm going to try and fix this now in a somewhat general way. Even if the product unit is willing to add the namespace of the `Layout` folder to the `_Imports` file of the Blazor Web App project template, it won't take effect until .NET 9 (probably), so we'll be stuck with .NET 8 requiring versioned content. I'll see if I can address this in a version-agnostic way.

**Comment by bmeenehan** [comment]
Thank you very much
It does seem like that is the intended way for custom layouts to be applied. Just a few extra sentences in the docs would clear the matter up for people like myself poking around at the system.
